client/jetpack-connect/index.js

```diff
@@ -72,4 +72,5 @@ export default function() {
 	page( '/jetpack/sso/:siteId?/:ssoNonce?', controller.sso );
 	page( '/jetpack/sso/*', controller.sso );
 	page( '/jetpack/new', controller.newSite );
+	page( '/jetpack/new/*', controller.notFoundError );
```

Comment:
What happens for `/jetpack/new/*` will also happen for `/jetpack/bla` too - so I'd suggest having this route match any previously unmatched `/jetpack/*` - what do you think?